### DISCUSSION TITLE
Organizing Server Actions and Database Queries in a Next.js 13 AppDir project

### DISCUSSION THREAD

**Discussion by laurent512** [vercel/next.js#55908] [App Router]
Hello, 

I'm currently developing a SaaS application using Next.js 13 with Appdir for server actions and Prisma for ORM and Auth JS. In my application, users can perform CRUD actions, which essentially involve Prisma queries after authorization verification.

My server actions are organized within files located in the "server-actions" folder, and my queries are kept in the "database-queries" folder ; both folder situated at the root level next to the "/app" directory.

I've noticed that I often end up with similar naming conventions for both the queries and the server actions. For instance, I have a function called createNewProject() that is invoked from a client component button. This function is considered a server action, and it calls createNewProject_prisma(), which is located in the "database_queries" folder.

I believe there could be cleaner and more interesting approaches to naming conventions and folder/file organization. 
Do you have any advice or recommendations for best practices or documentation resources that can help me improve this aspect of my project?

Thank you in advance

**Comment by leandronn** [comment]
Similar doubt here. how should I organize actions inside a page? for example: 

```jsx
export default async function MyProfile() {
  const session = await getServerSession(authOptions);

  async function getUser(userUuid) {
    'use server'
    const user = await prisma.users.findUnique({
      where: { uuid: userUuid },
    });

    if (!user) return 'no data';
    return user;
  };

  const user = await getUser(session.user?.id);
  
  return ({/*page return here*/});
  }
```

I need to add to that component more queries hence more server actions. Should I create a file in the same folder as page.js? Is this a good approach? Thanks in advance.
